gwctl/pkg/printer/gatewayclasses_test.go

```diff
@@ -296,3 +296,102 @@ foo-com-internal-gateway-class  foo-com-internal-gateway-class/controller  True
 		t.Errorf("Unexpected diff\ngot=\n%v\nwant=\n%v\ndiff (-want +got)=\n%v", got, want, diff)
 	}
 }
+
+// TestGatewayClassesPrinterDescribe_LabelSelector Tests label selector filtering for GatewayClasses in 'describe' command.
+func TestGatewayClassesPrinterDescribe_LabelSelector(t *testing.T) {
```

Comment:
Thanks for the thorough tests @yeedove!

I was actually thinking about this part and I feel that maybe testing LabelSelector is something which is better handled through unit tests within `discoverer_test.go` which will test functions like `fetchGatewayClass` (i.e. `TestFetchGatewayClass`). Idea being, I really want unit tests to be focused on functionality provided by an independent package.

The problem with LabelSelector tests right now is that they aren't just testing LabelSelector but are also testing how exactly the entire GatewayClass object gets printed (which is something that is already covered by the normal test functions like `TestGatewayClassesPrinter_PrintDescribeView`)

(I know that tests like `TestGatewayClassesPrinter_PrintDescribeView` are also not close to perfect since they too are indirectly testing the `resourcediscovery` package by relying on `DiscoverResourcesForGatewayClass` instead of actually constructing the entire ResourceModel itself. This is something that I would like to correct in the future if possible)

Within `discoverer_test.go`, these tests would only check that "given this label selector was provided, did I receive the correct GatewayClass names?"

Additionally, we could then also get rid of the LabelSelector tests for the `gwctl get`.

Reply:
- I recommend finalizing the label functionality for `get` and `describe` commands before optimization to avoid potential issues. For instance, the `get backend -l bar` command needs to display `REFERRED BY ROUTES` to these backends, but the relationship between HTTPRoutes and Services is not affected by labels, requiring additional handling. There may be better way :)
﻿
- In creating unit tests for `describe -l`, I found that the tests were overly dependent on the command’s output. As we plan to `Expand output for gwctl describe resources`, updating tests for every change in output is not good idea. I will focus on correctly verifying GatewayClass names instead.